### ISSUE TITLE
error message

### ISSUE THREAD

**Issue by robgpeak** [robgpeak/CL-WebExtension#35]
Hey @AddisonFreeman I just happened to be looking into my extension for something else and saw this.
<img width="793" alt="Screen Shot 2019-10-11 at 5 01 58 PM" src="https://user-images.githubusercontent.com/28393235/66684867-01e40800-ec49-11e9-8b44-3730bbbbc8a3.png">

**Comment by AddisonFreeman** [comment]
Okay, I’ll look into this tomorrow.

Addison Freeman

**Comment by AddisonFreeman** [comment]
@cgladue I believe this is a server side issue. 

Chrome needs the attributes listed above for any cookies sent in headers from the server. I think this error message appears on every endpoint call.

https://www.chromestatus.com/feature/5633521622188032
https://stackoverflow.com/questions/58191969/how-to-fix-set-samesite-cookie-to-none-warning-chrome-extension

**Comment by AddisonFreeman** [comment]
@cgladue Are you still around to comment on this thread? 
We're still seeing this issue and I believe it's a server side thing since I'm only reading cookie data set by the request header.

In addition to the Secure attribute, Samesite=none should be added to all set-cookie parameters.

<img width="1309" alt="Screen Shot 2019-11-05 at 11 36 54 AM" src="https://user-images.githubusercontent.com/5950267/68231495-cbd44280-ffc0-11e9-9488-146e1cfa6418.png">

Also, I don't know where (if?) the http:// call is coming from. 
<img width="684" alt="Screen Shot 2019-11-05 at 11 30 17 AM" src="https://user-images.githubusercontent.com/5950267/68230993-ce826800-ffbf-11e9-9915-96dc1ed665de.png">

**Comment by cgladue** [comment]
From Microsoft Blog:

> Google is now updating the standard and implementing their proposed changes in an upcoming version of Chrome. The change adds a new SameSite value, “None”, and changes the default behavior to “Lax”. This breaks OpenIdConnect logins, and potentially other features your web site may rely on, these features will have to use cookies whose SameSite property is set to a value of “None”. However browsers which adhere to the original standard and are unaware of the new value have a different behavior to browsers which use the new standard as the SameSite standard states that if a browser sees a value for SameSite it does not understand it should treat that value as “Strict”. This means your .NET website will now have to add user agent sniffing to decide whether you send the new None value, or not send the attribute at all.
> 
> .NET will issue updates to change the behavior of its SameSite attribute behavior in .NET 4.7.2 and in .NET Core 2.1 and above to reflect Google’s introduction of a new value. The updates for the .NET Framework will be available on November 19th as an optional update via Microsoft Update and WSUS if you use the “Check for Update” functionality. On December 10th it will become widely available and appear in Microsoft Update without you having to specifically check for updates. .NET Core updates will be available with .NET Core 3.1 starting with preview 1, in November.
> 
> .NET Core 3.1 will contain an updated enum definition, SameSite.Unspecified which will not set the SameSite property.

Once the update is available on november 19th, i will install on our web servers, we are already targeting .NET Framework 4.7.2 so it should automatically take effect after that.

**Comment by robgpeak** [comment]
@cgladue I just got off the phone with Addison and he said this might be a work around for now in order to get approval from Google, and to make it look like the extension is still somewhat working for existing users...

In essence we need the list of the top 100 URL's (most shopping trips, descending) in the following format "https://*.bestbuy.com/*","https://*.walmart.com/*", so he can copy and past them into the extension file.

https://www.loom.com/share/4133c3679684460f8fbd3a89591b34e5

**Comment by AddisonFreeman** [comment]
To add to this, Rob, you can upload the latest extension version now to get it “working” sooner.

**Comment by cgladue** [comment]
from what i understand this is just a warning its not going to be an issue before Chrome 80 is released in Feb or March next year ... so if theres an issue currently i dont think its because of this "SameSite" change google is making ...

**Comment by AddisonFreeman** [comment]
Watched the video from the dentist waiting room and can verify your instructions.

**Comment by AddisonFreeman** [comment]
Sorry, Chad. This is a *separate* issue. Right now the extension is down for all users. It’s a chrome extension site permissions issue.

**Comment by robgpeak** [comment]
> To add to this, Rob, you can upload the latest extension version now to get it “working” sooner.
> […](#)
> On Nov 5, 2019, 12:44 PM -0600, robgpeak ***@***.***>, wrote: @cgladue I just got off the phone with Addison and he said this might be a work around for now in order to get approval from Google, and to make it look like the extension is still somewhat working for existing users... In essence we need the list of the top 100 URL's (most shopping trips, descending) in the following format "https://.bestbuy.com/","https://.walmart.com/", so he can copy and past them into the extension file. https://www.loom.com/share/4133c3679684460f8fbd3a89591b34e5 — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

@AddisonFreeman I can't do this right now, just about to get on a conference call.

**Comment by AddisonFreeman** [comment]
Unfortunately I’m at the dentist. But in the meantime you can provide a list of the most popular advertisers to enable so we minimize the perception of downtime. Even 50% working is better than totally down.

I’m looking into a workaround for full dynamic coverage of advertisers but this will get us back up more quickly if only partially.

**Comment by robgpeak** [comment]
My call is actually at 3:00 not 2:00 like I thought it was.

@AddisonFreeman keep in mind that I already submitted the extension for review prior and this is the message Im seeing now.  So I am not even sure it would work if we resubmitted it now with your proposed fixes.

<img width="859" alt="Screen Shot 2019-11-05 at 2 08 20 PM" src="https://user-images.githubusercontent.com/28393235/68238147-90447300-ffd6-11e9-924e-a81f2d7a8aed.png">

**Comment by cgladue** [comment]
ok ill get a list

**Comment by AddisonFreeman** [comment]
Darn. That’s unfortunate. I wonder if there’s anything you can communicate to them to expedite the process.

**Comment by cgladue** [comment]
[mosttrips.zip](https://github.com/robgpeak/CL-WebExtension/files/3810828/mosttrips.zip)

**Comment by cgladue** [comment]
there it is in decending order by most shopping trips.

**Comment by robgpeak** [comment]
> Darn. That’s unfortunate. I wonder if there’s anything you can communicate to them to expedite the process.
> […](#)
> On Nov 5, 2019, 1:16 PM -0600, cgladue ***@***.***>, wrote: ok ill get a list — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

I think there is, but just let me know when you get back from the dentist and build the final version of the file including the URLs from Chads list and I will cancel and resubmit

**Comment by AddisonFreeman** [comment]
“If you unpublish you won’t be able to resubmit until your current item is reviewed”

That doesn’t sound very flexible.

**Comment by robgpeak** [comment]
> [mosttrips.zip](https://github.com/robgpeak/CL-WebExtension/files/3810828/mosttrips.zip)

@cgladue Please do me a favor and send another list that includes ALL "Active" advertisers sorted ascending from oldest to newest (more recently added).  This way if we add a new advertiser in the future, it would be easier to append the new advertiser URLs to the list for Addison to update the manifest.

Addison and I just spoke on the phone and I now have a much better idea of how this affects out extension.  We just need to add ALL the "possible" advertiser's URLs in the manifest, but from the server side, it will only display those that are "Extension Enabled = True", so we will be totally covered.

@AddisonFreeman when you get the list, please update the manifest and send me the completed version with the new version number so I can update the extension.

**Comment by AddisonFreeman** [comment]
@cgladue Following up on this. I'm ready to implement as soon as you send the list over.

**Comment by cgladue** [comment]
@robgpeak i dont track when they were "added" i only track the last time they were "updated" which should be once every 24 hours unless they are no longer valid advertisers

**Comment by AddisonFreeman** [comment]
As long as you send me a complete list every time I can tell which ones are new. Rob will know when new ones are added to prompt a change.

**Comment by cgladue** [comment]
ok, then the last list i sent you was a complete list

**Comment by robgpeak** [comment]
Then leave everything as it is.

>

**Comment by AddisonFreeman** [comment]
I don't think it was. Here's an example.
<img width="650" alt="Screen Shot 2019-11-11 at 2 07 09 PM" src="https://user-images.githubusercontent.com/5950267/68617326-a649ac00-048c-11ea-9289-fe2b347580b5.png">

**Comment by robgpeak** [comment]
an example of what?  It seems to be working properly in the example you've shown.

**Comment by AddisonFreeman** [comment]
No, that site isn’t on the list so the notification doesn’t show when you click on it.

**Comment by robgpeak** [comment]
It worked for me when I clicked through.  But I get your point that if they are NOT enabled, their banner will still show up above their google listing, but the pop up on the site itself will not show up.

<img width="1141" alt="Screen Shot 2019-11-11 at 3 25 50 PM" src="https://user-images.githubusercontent.com/28393235/68618633-c2067f80-0497-11ea-8d50-730e398ddfd7.png">
